src/AppInstallerCLICore/ExecutionContext.cpp

```diff
@@ -197,6 +197,8 @@ namespace AppInstaller::CLI::Execution
 
     void Context::SetExecutionStage(Workflow::ExecutionStage stage, bool allowBackward)
     {
+        Logging::SetExecutionStage(static_cast<uint32_t>(stage));

```

Comment:
>Logging::SetExecutionStage(static_cast<uint32_t>(stage)); [](http://example.com/codeflow?start=8&length=57)

this should probably go to the end, after error checking

Reply:
Thinking was that telemetry should reflect the reality as is even in error cases. But I guess we are more interested in knowing the last valid execution stage from failure telemetry events rather than the invalid execution stage it is asked to set to.

Reply:
yep, we care more about last valid stage so I think it should be after error checking?